dwave/optimization/model.pyi

```diff
@@ -12,29 +12,166 @@
 #    See the License for the specific language governing permissions and
 #    limitations under the License.
 
+import tempfile
+import typing
+
+import numpy
 import numpy.typing
 
+from dwave.optimization.symbols import *
+
 
-class ArraySymbol:
-    state: StatesView
+_ShapeLike: typing.TypeAlias = typing.Union[int, typing.Sequence[int]]
 
 
 class Model:
-    def constant(self, array_like) -> Constant: ...
+    def __init__(self): ...
+
+    @property
+    def objective(self) -> ArraySymbol: ...
+    @property
+    def states(self) -> States: ...
+
+    def add_constraint(self, value: ArraySymbol): ...
+    def binary(self, shape: typing.Optional[_ShapeLike] = None) -> BinaryVariable: ...
+    def constant(self, array_like: numpy.typing.ArrayLike) -> Constant: ...
+    def decision_state_size(self) -> int: ...
+
+    def disjoint_bit_sets(
+        self, primary_set_size: int, num_disjoint_sets: int,
+        ) -> typing.Tuple[DisjointBitSets, typing.Tuple[DisjointBitSet, ...]]: ...
+
+    def disjoint_lists(
+        self, primary_set_size: int, num_disjoint_lists: int,
+        ) -> typing.Tuple[DisjointLists, typing.Tuple[DisjointList, ...]]: ...
+
+    @classmethod
+    def from_file(
+        cls,
+        file: typing.Union[typing.BinaryIO, typing.ByteString, str],
+        *,
+        check_header: bool = True,
+        ) -> Model: ...
+
+    def integer(
+        self,
+        shape: typing.Optional[_ShapeLike] = None,
+        lower_bound: typing.Optional[int] = None,
+        upper_bound: typing.Optional[int] = None,
+        ) -> IntegerVariable: ...
+
+    def into_file(
+        self,
+        file: typing.Union[typing.BinaryIO, typing.ByteString, str],
+        *,
+        max_num_states: int = 0,
+        only_decision: bool = False,
+        ): ...
```

Comment:
should this have a typing `->` typing.BinaryIO: ...` here?

Reply:
`to_file()` returns a file, but `into_file()` does not.

Reply:
makes sense